From 5651b3afc5a9173670907c87e2bc46df9d3f29fb Mon Sep 17 00:00:00 2001
From: Raju Rangoju <Raju.Rangoju@amd.com>
Date: Tue, 8 Feb 2022 12:39:24 +0530
Subject: [PATCH 11/48] i2c: amd-mp2: avoid using pci_intx if msi or msix is
 supported

Avoid the usage of pci_intx if the harware supports either msi or msix.

Signed-off-by: Raju Rangoju <Raju.Rangoju@amd.com>
Signed-off-by: Sudheesh Mavila <sudheesh.mavila@amd.com>
Signed-off-by: rgaridap <Ramesh.Garidapuri@amd.com>
Change-Id: Ic3f616def0a47c427f82adf9d5a3a1286ef035f0
---
 drivers/i2c/busses/i2c-amd-mp2-pci.c | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/drivers/i2c/busses/i2c-amd-mp2-pci.c b/drivers/i2c/busses/i2c-amd-mp2-pci.c
index ce130a821ea5..a72c24f19853 100644
--- a/drivers/i2c/busses/i2c-amd-mp2-pci.c
+++ b/drivers/i2c/busses/i2c-amd-mp2-pci.c
@@ -314,14 +314,18 @@ static int amd_mp2_pci_init(struct amd_mp2_dev *privdata,
 			goto err_dma_mask;
 	}
 
-	/* Set up intx irq */
+	/* Set up MSI-x irq */
 	writel(0, privdata->mmio + AMD_P2C_MSG_INTEN);
-	pci_intx(pci_dev, 1);
-	rc = devm_request_irq(&pci_dev->dev, pci_dev->irq, amd_mp2_irq_isr,
-			      IRQF_SHARED, dev_name(&pci_dev->dev), privdata);
+	rc = pci_alloc_irq_vectors(pci_dev, 1, 1, PCI_IRQ_MSIX);
+	if (rc < 0) {
+		dev_err(&pci_dev->dev, "Failed to allocate MSI-x interrupts err=%d\n", rc);
+		return rc;
+	}
+	rc = devm_request_irq(&pci_dev->dev, pci_irq_vector(pci_dev, 0),
+			      amd_mp2_irq_isr, 0, dev_name(&pci_dev->dev), privdata);
 	if (rc)
 		pci_err(pci_dev, "Failure requesting irq %i: %d\n",
-			pci_dev->irq, rc);
+			pci_irq_vector(pci_dev, 0), rc);
 
 	return rc;
 
-- 
2.27.0

